plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnector.java

```diff
@@ -660,53 +605,46 @@ public void startExec(String execId, LogMessageProcessor execOutputProcessor) th
-        try {
-            final DockerResponse response = connection.method("GET").path(String.format("/exec/%s/json", execId)).request();
-
+        try (DockerConnection connection = openConnection(dockerDaemonUri).method("GET")
+                                                                          .path("/exec/" + execId + "/json")) {
+            final DockerResponse response = connection.request();
             final int status = response.getStatus();
             if (200 != status) {
-                final String msg = CharStreams.toString(new InputStreamReader(response.getInputStream()));
-                throw new DockerException(String.format("Error response from docker API, status: %d, message: %s", status, msg), status);
+                throw new DockerException(getDockerExceptionMessage(response), status);
             }
-            return JsonHelper.fromJson(response.getInputStream(), ExecInfo.class, null, FIRST_LETTER_LOWERCASE);
+            return parseResponseStreamAndClose(response.getInputStream(), ExecInfo.class);
         } catch (Exception e) {
-            throw new IOException(e.getMessage(), e);
-        } finally {
-            connection.close();
+            throw new IOException(e.getLocalizedMessage(), e);
         }
     }
 
-
     public ContainerProcesses top(String container, String... psArgs) throws IOException {
-        final DockerConnection connection = openConnection(dockerDaemonUri);
-        try {
-            final List<Pair<String, ?>> headers = new ArrayList<>(2);
-            headers.add(Pair.of("Content-Type", "text/plain"));
-            headers.add(Pair.of("Content-Length", 0));
-            final String path;
-            if (psArgs == null || psArgs.length == 0) {
-                path = String.format("/containers/%s/top", container);
-            } else {
-                final StringBuilder pathBuilder = new StringBuilder();
-                pathBuilder.append("/containers/").append(container).append("/top?ps_args=");
-                for (int i = 0, l = psArgs.length; i < l; i++) {
-                    if (i > 0) {
-                        pathBuilder.append('+');
-                    }
-                    pathBuilder.append(URLEncoder.encode(psArgs[i], "UTF-8"));
+        final List<Pair<String, ?>> headers = new ArrayList<>(2);
+        headers.add(Pair.of("Content-Type", "text/plain"));
+        headers.add(Pair.of("Content-Length", 0));
+        final DockerConnection connection = openConnection(dockerDaemonUri).method("GET")
+                                                                           .path("/containers/" + container + "/top")
+                                                                           .headers(headers);
+        if (psArgs != null && psArgs.length != 0) {
+            StringBuilder psArgsQueryBuilder = new StringBuilder();
+            for (int i = 0, l = psArgs.length; i < l; i++) {
```

Comment:
URLEncoder.encode throws UnsupportedEncodingException, so lambda won't be pretty. WDYT?